src/test/java/com/arpnetworking/commons/math/AccumulatorTest.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2020 Dropbox
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.commons.math;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Supplier;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for the {@link Accumulator} implementations.
+ *
+ * @author Ville Koskela (ville at koskilabs dot com)
+ */
+@RunWith(Parameterized.class)
+public final class AccumulatorTest {
+
+    private static final List<Double> RANDOM_VALUE_DATA_SET;
+    private static final List<Double> RANDOM_CANCEL_DATA_SET;
+    private static final Logger LOGGER = LoggerFactory.getLogger(AccumulatorTest.class);
+
+    static {
+        // Random values data set
+        final List<Double> valueDataSet = new ArrayList<>();
+        final Random valueRandom = new Random(564564359);
+        final double max = 1.0e16;
+        final double min = -1.0e16;
+        for (int i = 0; i < 4000000; ++i) {
+            valueDataSet.add(valueRandom.nextDouble() * (max - min) + min);
+        }
+        RANDOM_VALUE_DATA_SET = Collections.unmodifiableList(valueDataSet);
+
+        // Random cancellations data set
```

Comment:
What are `cancellations` in this context?

Is it because you have two magnitudes of numbers which pairwise sum to zero?